jsettlers.tools/src/main/java/jsettlers/buildingcreator/editor/BuildingCreatorApp.java

```diff
@@ -117,24 +152,91 @@ private IMapInterfaceConnector startMapWindow() throws JSettlersLookAndFeelExecp
 		return SwingManagedJSettlers.showJSettlers(new FakeMapGame(map));
 	}
 
-	private JButton createToolChangeBar() {
-		JButton button = new JButton("Select tool...");
-		button.addActionListener(e -> {
-			ToolType newTool = (ToolType) JOptionPane.showInputDialog(null, "Select building type", "Building Type",
-					JOptionPane.QUESTION_MESSAGE, null, ToolType.values(), tool);
-
-			if (newTool != null) {
-				tool = newTool;
-			}
-
-			for (int x = 0; x < map.getWidth(); x++) {
-				for (int y = 0; y < map.getWidth(); y++) {
-					reloadColor(new ShortPoint2D(x, y));
-				}
-			}
-		});
-		return button;
+        private void occupyBulding() {
+            log.debug("occupyBuilding");
+            
+            BuildingVariant variant = definition.getBuilding();
+            List<IBuildingOccupier> occupiers = null;
+            IBuilding.IOccupied building = (IBuilding.IOccupied)map.getBuilding();
+            occupiers = (List<IBuildingOccupier>)building.getOccupiers();
+            log.debug("OccupierPlaces:");
+            for (OccupierPlace op: variant.getOccupierPlaces()) {
```

Comment:
This should be a member function of the building. You shouldn't edit the state (getOccupiers()) of the building outside of the building

Reply:
JSettlers generally implements the following pattern:
There is one interface that describes the in game visible state of an object (like IGraphicsMovable) and there is a class that implements the game logic (like Movable).
But you can also add alternative implementation of the game logic or dummy implementations that only show a constant state (like BuildingCreatorGraphicsMovable).

PseudoBuilding does the same for buildings as BuildingCreatorGraphicsMovable does for movables.
You can just modify the logic behind its state as the needs are for the BuildingCreatorApp.

Reply:
So it would make sense to have methods like occupy() and evacuate() in IOccupied?
But how does the game today resolve the same issue? The methods do not exist today.

Reply:
Oh no. IOccupied is just there to query what the user can see and **not** how this can change. This is purely part of the game logic and not the graphics interface.
occupy/evacuate should be functions of PseudoBuilding and not IOccupied

Reply:
Ok, I now understood what you actually pointed at. Now not IOccupied is modified but just the PseudoBuilding.